### PR TITLE
feat(dataset): support basic version control methods for "Dataset"

### PR DESCRIPTION
**Pull request by graczhual**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2ccta9w 实现 Dataset 类中的 checkout 和 status 方法，支持基础的版本管理功能](https://app.clickup.com/t/2ccta9w)

File: graviti/dataset/dataset.py [hunk 1]
```diff
@@ -243,6 +256,15 @@ def checkout(self, revision: str) -> None:
                 the branch, or the tag.
 
         """
+        try:
+            branch = self.branches.get(revision)
```
Comment (linjiX) [inline graviti/dataset/dataset.py:260]: what if the input revision is a tag?

Comment (graczhual) [reply graviti/dataset/dataset.py:260]: the revision in `self.commits.get(revision)` could be the tag name

**Review by linjiX — APPROVED** [review]

File: graviti/dataset/dataset.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 from typing import Any, Dict, Iterator, KeysView, Mapping, Optional
 
-from tensorbay.client.status import Status
+from tensorbay.exception import ResourceNotExistError
```
Comment (linjiX) [inline graviti/dataset/dataset.py:10]: commit message:
`function` -> `methods`?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/manager/branch.py [hunk 3]
```diff
@@ -46,18 +65,32 @@ def get(self, name: str) -> Branch:
         Arguments:
             name: The required branch name.
 
-        Return:
+        Raises:
+            TypeError: When the given branch is illegal.
```
Comment (Lee-000) [inline graviti/manager/branch.py:69]: `the given branch name`?

File: graviti/manager/commit.py [hunk 4]
```diff
@@ -23,7 +26,24 @@ class CommitManager:
     """
 
     def __init__(self, dataset: "Dataset") -> None:
-        pass
+        self._dataset = dataset
+
+    def _generate(
+        self, revision: str, offset: int = 0, limit: int = 128
```
Comment (Lee-000) [inline graviti/manager/commit.py:32]: Should there be a default value for `revision`?

File: graviti/dataset/dataset.py [hunk 5]
```diff
@@ -43,15 +44,18 @@ def __init__(
         dataset_id: str,
         name: str,
         *,
-        status: Status,
         alias: str,
+        branch_name: Optional[str] = None,
+        commit_id: str,
```
Comment (Lee-000) [inline graviti/dataset/dataset.py:49]: Should the param with a default value be at the last?

File: graviti/manager/commit.py [hunk 6]
```diff
@@ -32,20 +52,50 @@ def get(self, revision: Optional[str] = None) -> Commit:
...
+            ResourceNotExistError: When the required commit does not exist.
+
+        Returns:
             The :class:`.Commit` instance with the given revision.
 
         """
+        if revision is None:
+            revision = self._dataset.commit_id
+
+        if not revision:
```
Comment (Lee-000) [inline graviti/manager/commit.py:66]: In what case would the `commit_id` be empty?

**Review by Lee-000 — CHANGES_REQUESTED** [review]

File: graviti/manager/commit.py (hunk 6 above)
Comment (graczhual) [reply graviti/manager/commit.py:66]: if `commit_id=""`, the OpenAPI will return all commits.

Comment (graczhual) [reply graviti/manager/commit.py:66]: if `commit_id=""`, the OpenAPI will return all commits.

**Review by linjiX — APPROVED** [review]

**Review by Lee-000 — APPROVED** [review]

**Pull request merged**